httpbakery/agent/agent.go

```diff
@@ -0,0 +1,161 @@
+/*
+Package agent implements the agent login mechanism. The agent login
+mechanism provides a form of interaction that can be used in
+situations where interaction with the user is not possible.
+
+An agent login works as follows:
+
+	    Agent                            Login Service
+	      |                                    |
+	      | GET visitURL with agent cookie     |
+	      |----------------------------------->|
+	      |                                    |
+	      |    Macaroon with local third-party | 
+	      |                             caveat |
+	      |<-----------------------------------|
+	      |                                    |
+	      | GET visitURL with agent cookie &   |
+	      | discharged macaroon                |
+	      |----------------------------------->|
+	      |                                    |
+	      |               Agent login response |
+	      |<-----------------------------------|
+	      |                                    |
+
+The agent cookie is a cookie named "agent-login" its value is a base64
```

Comment:
The agent cookie is a cookie named "agent-login" holding a base64-encoded JSON object of the form:

?